extension/goext/schemas.go

```diff
@@ -158,6 +158,22 @@ type ISchema interface {
 	RegisterRawType(rawResourceType interface{})
 
 	RawSchema() interface{}
+
+	// DerivedSchemas returns list of schemas that extend schema with given id
+	DerivedSchemas() []ISchema
+
+	// MakeColumns generates an array that has Gohan style colmun names
+	MakeColumns() []string
```

Comment:
I don't like name "MakeColumns" since this doesn't make any columns. It's a read operation so word 'make' seems wrong. It's not really tied to Schema, so maybe introduce free function, not as interface method.

Reply:
This function is called after the function from the "sql" package with the same name, which uses raw schema, so that's why it's a part of schema interface.

Reply:
How does it relate to Properties?

Reply:
Properties is another function that returns properties of given schema.

Reply:
Fact that function name in `sql` package is wrong, doesn't implies that new code should follow it. I meant that this free function could take schema as parameter or take list Properties to iterate over.

Reply:
Ok, we can't make it a free function. Just change name to something else. 'MakeColumns' seems not right.